src/components/session-list.tsx

```diff
@@ -0,0 +1,147 @@
+"use client";
+
+import { useEffect, useState, SVGProps } from "react";
+import { collection, getDocs, query, orderBy } from "firebase/firestore";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import { db } from "@/firebase/firebaseConfig";
+import { useRouter } from "next/navigation";
+
+export function SessionList() {
+  const [sessions, setSessions] = useState<
+    { id: string; sessionName: string; queueCount: number }[]
+  >([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const router = useRouter(); // Initialize the router
+
+  useEffect(() => {
+    async function fetchSessions() {
+      try {
+        const sessionsCollection = collection(db, "sessions");
+        const sessionsQuery = query(sessionsCollection, orderBy("createdAt"));
+        const sessionSnapshot = await getDocs(sessionsQuery);
+
+        // Fetch queue counts for each session
+        const sessionList = await Promise.all(
+          sessionSnapshot.docs.map(async (doc) => {
+            const sessionName = doc.data().sessionName || "Unnamed Session";
+
+            // Fetch the queue subcollection for this session
+            const queueCollection = collection(db, "sessions", doc.id, "queue");
+            const queueSnapshot = await getDocs(queueCollection);
+            const queueCount = queueSnapshot.size;
+
+            return {
+              id: doc.id,
+              sessionName,
+              queueCount,
+            };
+          })
+        );
+
+        setSessions(sessionList);
+        setLoading(false);
+      } catch (err) {
+        console.error("Error fetching sessions: ", err);
+        setError("Failed to load sessions. Please try again later.");
+        setLoading(false);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Fetching data from Firestore is done in a nested manner. First, you fetch all sessions, then for each session, you fetch its queue. This approach can lead to performance issues when the number of sessions increases because it results in multiple network requests. Instead, consider storing the `queueCount` directly in the session document in Firestore. This way, you only need to fetch the sessions, and the `queueCount` will be included in each session's data. You can update the `queueCount` whenever a participant is added or removed from the queue.

```diff
-        // Fetch queue counts for each session
-        const sessionList = await Promise.all(
-          sessionSnapshot.docs.map(async (doc) => {
-            const sessionName = doc.data().sessionName || "Unnamed Session";
-
-            // Fetch the queue subcollection for this session
-            const queueCollection = collection(db, "sessions", doc.id, "queue");
-            const queueSnapshot = await getDocs(queueCollection);
-            const queueCount = queueSnapshot.size;
-
-            return {
-              id: doc.id,
-              sessionName,
-              queueCount,
-            };
-          })
-        );
+        const sessionList = sessionSnapshot.docs.map((doc) => {
+          const data = doc.data();
+          const sessionName = data.sessionName || "Unnamed Session";
+          const queueCount = data.queueCount || 0;
+
+          return {
+            id: doc.id,
+            sessionName,
+            queueCount,
+          };
+        });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
@coderabbitai Suggest how to implement the recommendation